apm-sniffer/apm-agent-core/src/main/java/org/skywalking/apm/agent/core/plugin/PluginCfg.java

```diff
@@ -29,6 +44,55 @@ void load(InputStream input) throws IOException {
     }
 
     public List<String> getPluginClassList() {
-        return pluginClassList;
+        List<String> pluginClasses = new ArrayList<String>();
+        for (PluginDefine plugin : pluginClassList) {
+            pluginClasses.add(plugin.defineClass);
+        }
+        return pluginClasses;
+    }
+
+    private static class PluginDefine {
+
+
+        private static final PluginDefine NOOP_PLUGIN = new PluginDefine() {
+            @Override public boolean disabled(List<String> disablePlugins) {
+                return true;
+            }
+        };
+        /**
+         * Plugin name.
+         */
+        String name;
+
+        /**
+         * The class name of plugin defined.
+         */
+        String defineClass;
+
+        private PluginDefine() {
+        }
+
+        private PluginDefine(String name, String defineClass) {
+            this.name = name;
+            this.defineClass = defineClass;
+        }
+
+        public static PluginDefine build(String define) {
+            if (StringUtil.isEmpty(define)) {
+                return NOOP_PLUGIN;
```

Comment:
Did not like the design of retuning noop plugin define. Hard to understand and make no sense.